src/search/landmarks/landmark_heuristic.cc

```diff
@@ -61,6 +61,15 @@ void LandmarkHeuristic::initialize(const plugins::Options &opts) {
         *lm_graph, opts.get<bool>("prog_goal"),
         opts.get<bool>("prog_gn"), opts.get<bool>("prog_r"));
 
+    /* TODO: Maybe deal with this differently, e.g., mark initial state
+        as dead-end. (Similar use case in M&S maybe?) */
+    if (landmark_graph_has_cycle_of_natural_orderings()) {
+        log << "Found a cycle of natural (or stronger) landmark orderings. By "
+            << "the definition of natural orderings, the problem is unsolvable."
+            << endl;
+        utils::exit_with(utils::ExitCode::SEARCH_UNSOLVABLE);
+    }
```

Comment:
When working on the implementation draft, we asked ourselves whether we should exit immediately or do it some other way. The reason we went with this option is because we found a similar usage of `ExitCode::SEARCH_UNSOLVABLE` in `pdbs/cegar.cc`. I don't think we should add more of these if we don't want them in the long term and I see your points why we don't want them. Using a flag instead is a straightforward alternative that I can implement right away.

Reply:
Only because there is a cycle in the initial state doesn't mean the heuristic value is infinite for *every* state. I'm not sure whether this is relevant or not, but returning `DEAD_END` in `compute_heuristic(...)` for arbitrary states if there is a cycle of natural orderings in the initial state feels wrong to me. But was this your suggestion, anyway? Happy to implement it this way even if I'm not the biggest fan of the solution.

Reply:
The discussion is perhaps complex enough that I suggest we talk on Zoom. Are you available today some time between 16:45 and 17:15? In this case, can you call me on Zoom?

Reply:
We've had a discussion offline. I've implemented the flag-approach and added a somewhat extensive comment to reflect our discussion. Maybe it is unnecessary or too verbose, I don't know. I'm also fine with just dropping it, but I thought it might make sense to document somewhere what we discussed.

Reply:
I had a look at the comment and like it.

Reply:
Thanks! :)